src/ResourceManager/Profile/Commands.Profile/Utilities/EnvironmentHelper.cs

```diff
@@ -0,0 +1,117 @@
+            {
+                if (armEndpoint.Contains("http://"))
+                {
+                    armEndpoint = armEndpoint.Substring(7);
+                    armEndpoint = "https://" + armEndpoint;
+                }
+                else
+                {
+                    armEndpoint = "https://" + armEndpoint;
+                }
+            }
+
+            armEndpoint = armEndpoint.TrimEnd('/');
+
+            return armEndpoint;
+        }
+
+        /// <summary>
+        /// Retrieves the domain.
+        /// </summary>
+        /// <param name="portalEndpoint">The portal endpoint.</param>
+        /// <returns>Domain</returns>
+        internal static string RetrieveDomain(string portalEndpoint)
+        {
+            return portalEndpoint.Replace(portalEndpoint.Split('.')[0], "").TrimEnd('/').TrimStart('.');
+        }
+
+        /// <summary>
+        /// Retrieves the meta data endpoints.
+        /// </summary>
+        /// <param name="url">The URL.</param>
+        /// <returns></returns>
+        internal static async Task<MetadataResponse> RetrieveMetaDataEndpoints(string url)
+        {
+            string validUrl = GetValidEndpoint(url);
+            validUrl = String.Concat(validUrl.ToLower(), "/metadata/endpoints?api-version=1.0");
+            MetadataResponse response = null;
+            using (HttpClient client = new HttpClient())
+            {
+                // Create HTTP transport objects
+                HttpRequestMessage httpRequest = null;
+                try
+                {
+                    httpRequest = new HttpRequestMessage
+                                  {
+                                      Method = HttpMethod.Get,
+                                      RequestUri = new Uri(validUrl)
+                                  };
+
+                    // Send Request
+                    HttpResponseMessage httpResponse = null;
+                    try
+                    {
+                        httpResponse = await client.SendAsync(httpRequest).ConfigureAwait(false);
+                        HttpStatusCode statusCode = httpResponse.StatusCode;
+                        if (statusCode != HttpStatusCode.OK)
+                        {
+                            CloudException ex = CloudException.Create(httpRequest, null, httpResponse, await httpResponse.Content.ReadAsStringAsync().ConfigureAwait(false));
+                            throw ex;
+                        }
+
+                        if (statusCode == HttpStatusCode.OK)
+                        {
+                            string responseContent = await httpResponse.Content.ReadAsStringAsync().ConfigureAwait(false);
+                            response = JsonConvert.DeserializeObject<MetadataResponse>(responseContent);
+                        }
+                    }
+                    finally
+                    {
+                        httpResponse?.Dispose();
```

Comment:
Using only disposes httpclient in this case; not httprequest & httpresponse objects.